main/proc.sql

```diff
@@ -222,6 +208,75 @@ $$ LANGUAGE SQL;
 /**
 * CORE ROUTINES
 */
+--is end_hour necessary?
+CREATE OR REPLACE PROCEDURE approve_meeting (floor_no INTEGER, room_no INTEGER, date DATE, start_hour INTEGER, end_hour INTEGER, eid INTEGER)
+AS $$
+BEGIN
+	IF EXISTS
+		(SELECT 1
+		 FROM Manager m, Employees e, Sessions s, Meeting_rooms r
+		 WHERE eid = m.eid --is a manager
+		 --check whether manager's did same as room's did
+		 AND eid = e.eid
+		 AND e.did = r.did
+		 AND s.floor = r.floor
+		 AND s.room = r.room)
+		THEN
+			UPDATE Sessions
+			SET approving_manager_eid = eid
+			WHERE floor = floor_no
+			AND room = room_no;
+			/*AND date from time = date
+			AND hour from time >= start_hour
+			AND hour from time <= end_hour*/
+	END IF;
+END
+$$ LANGUAGE plpgsql;
+
+--is end_hour necessary?
+CREATE OR REPLACE PROCEDURE leave_meeting (floor_no INTEGER, room_no INTEGER, date DATE, start_hour INTEGER, end_hour INTEGER, eid INTEGER)
+AS $$
+BEGIN
+	IF EXISTS
+		(SELECT 1
+		 FROM Sessions
+		 WHERE approving_manager_eid ISNULL --meeting not approved yet
+		 --check whether eid is participating in the specified meeting
+		 AND participant_eid = eid 
+		 AND floor = floor_no
+		 AND room = room_no);
+		 /*AND date from time = date
+		 AND hour from time >= start_hour
+		 AND hour from time <= end_hour*/
+		THEN
+			DELETE FROM Sessions
+			WHERE participant_eid = eid
+			AND floor = floor_no
+			AND room = room_no;
+			/*AND date from time = date
+			AND hour from time >= start_hour
+			AND hour from time <= end_hour*/
+	END IF;
+END
+$$ LANGUAGE plpgsql;
+
+--need to consider contact tracing?
```

Comment:
no i got triggers for removing from meeting. moreover, there's the fever check you added